scripts/fix_sq_headers.py

```diff
@@ -0,0 +1,229 @@
+
+    sb.run(['samtools', 'reheader', '--no-PG', '--in-place', newhdr_file, incram], check=True)
+    os.rename(incram, outcram)
+    sb.run(['samtools', 'index', '-o', outcrai, outcram], check=True)
+
+    newref = classify_SQ(outcram)
+    print(f'{outcram}: reheadered: {newref}')
+    if newrefset is not None and newref != newrefset:
+        raise ValueError(f'{outcram}: unexpected reference headers written: {newref}')
+
+    return {
+        'cram_size': os.stat(outcram).st_size,
+    }
+
+
+def do_metamist_update(dataset, sequencing_type, seqgroup, oldpath, newpath, result):
+    analysis = models.Analysis(
+        type='cram',
+        status=models.AnalysisStatus('completed'),
+        output=str(newpath),
+        sequencing_group_ids=[seqgroup],
+        meta={
+            'sequencing_type': sequencing_type,
+            'size': result['cram_size'],
+            'source': f'Reheadered from {oldpath}',
+        },
+    )
+
+    aid = AnalysisApi().create_analysis(dataset, analysis)
+    print(f'Created Analysis(id={aid}, output={newpath}) in {dataset}')
+
+
+if __name__ == '__main__':
+    config = get_config(True)
+
+    dataset = config['workflow']['dataset']
+    if config['workflow']['access_level'] == 'test' and not dataset.endswith('-test'):
+        dataset = f'{dataset}-test'
+
+    if source_dir := config['workflow'].get('source', None):
+        print(f'Scanning {source_dir} for *.cram files')
+        cram_list = [fn for fn in to_path(source_dir).iterdir() if fn.suffix == '.cram']
+    else:
+        print(f'Finding CRAMs for {dataset} in database')
+        cram_list = query_metamist(dataset)
+
+    print(f'Total CRAM file entries found: {len(cram_list)}')
+
+    sequencing_type = config['workflow']['sequencing_type']
+    expected_refset = config['references']['expected']
+    new_reference_path = config['references']['new']
+    new_refset = config['references'].get('newset', None)
+    dry_run = config['workflow'].get('dry_run', False)
+
+    reheader_list = []
+    for fname in cram_list:
+        path = to_path(fname)
+        reheadered_path = path.parent / 'reheadered' / path.name
+
+        if not path.exists():
+            print(f'{fname}: file does not exist')
+        elif reheadered_path.exists():
+            print(f'{reheadered_path}: reheadered file already exists')
+        else:
+            reheader_list.append((path, reheadered_path, path.stat().st_size))
+
+    if len(reheader_list) == 0:
+        sys.exit('No CRAMs to be reheadered!')
+
+    print(f'CRAM files to be reheadered: {len(reheader_list)}')
+
+    b = get_batch(f'Reheader {dataset} to {new_reference_path}')
+    new_reference = b.read_input(new_reference_path)
+
+    for path, newpath, filesize in reheader_list:
+        j = b.new_python_job(f'Reheader {path}')
+        j.image(image_path('samtools'))
+        j.storage(filesize * 1.2)  # Allow some extra space for index file, references, etc
+
+        j_result = j.call(do_reheader,
```

Comment:
This doesn't feel like the regular black formatting. Can you pass it in using kwargs (eg: `dry_run=dry_run, ...`) for clarity?

Reply:
I could, but does it really matter for a ~200 line one-shot script under some time pressure?